### PR TITLE
[kube-1.6.x] remove dep and use vndr

### PR DESCRIPTION
**Pull request by runcom**
This PR removes dep and uses vndr to manage vendors in the kube-1.6.x branch - I'm going to update it on master as well - it's too easy to vendor the right vendors with **vndr** and it also clean up unused files (tests, readmes etc etc).

```sh
go get -u github.com/LK4D4/vndr
# to update just one vendor at a time
vndr github.com/somegithub/someproject
# to update all vendors in vendor.conf
vndr
```

@mrunalp @sameo PTAL

**Note this PR isn't running node-e2e tests! I need to add the correct playbook for it but it's still TODO**

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
I believe we need to use vndr:
```
+14,766 −4,053,924 
```
:smile:

**Comment by runcom** [comment]
the vndr conf is also pretty easy to understand and use https://github.com/kubernetes-incubator/cri-o/pull/558/files#diff-4061fcef378a6d912e14e2ce162a1995

**Comment by runcom** [comment]
I'm manually running kube tests, if you guys want as well, please do so.

File: vendor.conf [hunk 1]
```diff
@@ -0,0 +1,73 @@
+k8s.io/kubernetes v1.6.4 https://github.com/kubernetes/kubernetes
+# https://github.com/kubernetes/client-go#compatibility-matrix
+k8s.io/client-go v3.0.0-beta.0 https://github.com/kubernetes/client-go
+k8s.io/apimachinery release-1.6 https://github.com/kubernetes/apimachinery
+k8s.io/apiserver release-1.6 https://github.com/kubernetes/apiserver
+#
+github.com/Sirupsen/logrus master
+github.com/containers/image master
```
Comment (mrunalp) [inline vendor.conf]: Shouldn't we pin these to specific revisions? Otherwise, it may get updated while adding other deps.

Comment (runcom) [inline vendor.conf]: Yeah, I'll pin some of them, even if, if you add a new dep you don't need to run the full vndr again, just for that one you're adding.

**Comment by sameo** [comment]
I'm fine using a simpler and less immature tool. And I love the `git diff --stat` :) 
However, do we want to use it also for master ? Also, are we planning to eventually switch back to dep when it gets into the official golang distribution ?

**Comment by runcom** [comment]
> However, do we want to use it also for master ?

possibly, as you also point out, we would switch back to dep when it's actiually better :)

**Comment by runcom** [comment]
node-e2e and e2e ran on this w/o issues, after this is merged I'm going to add the playbook for the CI for kube-1.6.x

**Comment by runcom** [comment]
fixed Mrunal's comment, I'm going to take this in for kube-1.6.x

**Pull request merged**